ci/github-script/withRateLimit.js

```diff
@@ -0,0 +1,61 @@
+module.exports = async function ({ github, core }, callback) {
+  const Bottleneck = require('bottleneck')
+
+  const stats = {
+    issues: 0,
+    prs: 0,
+    requests: 0,
+    artifacts: 0,
+  }
+
+  // Rate-Limiting and Throttling, see for details:
+  //   https://github.com/octokit/octokit.js/issues/1069#throttling
+  //   https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api
+  const allLimits = new Bottleneck({
+    // Avoid concurrent requests
+    maxConcurrent: 1,
+    // Will be updated with first `updateReservoir()` call below.
+    reservoir: 0,
+  })
+  // Pause between mutative requests
+  const writeLimits = new Bottleneck({ minTime: 1000 }).chain(allLimits)
+  github.hook.wrap('request', async (request, options) => {
+    // Requests to the /rate_limit endpoint do not count against the rate limit.
+    if (options.url == '/rate_limit') return request(options)
+    // Search requests are in a different resource group, which allows 30 requests / minute.
+    // We do less than a handful each run, so not implementing throttling for now.
+    if (options.url.startsWith('/search/')) return request(options)
```

Comment:
If this is becoming a general use bottlenecking util, we'll probably end up breaking the "only an handful" assumption at some point.

Should we tweak the comment to be a TODO/FIXME style comment with that in mind?


```suggestion
    // Search requests are in a different resource group, which allows 30 requests / minute.
    // So far, none of our workflows do more than a handful each run, so not implementing throttling for now.
    // TODO: consider adding a separate reservoir for search endpoints.
    if (options.url.startsWith('/search/')) return request(options)
```

That said, this could become a slippery slope: if we had a separate reservoir for search, then maybe we'd want another reservoir for graphql, and other ones for anything else with a separate ratelimit...